src/main/java/org/fundacionjala/at15/katas/langtonant/diana/Grid.java

```diff
@@ -0,0 +1,63 @@
+package org.fundacionjala.at15.katas.langtonant.diana;
+
+public class Grid {
+    private int rows;
+    private int columns;
+    private int[][] grid;
+
+    public Grid(int rows, int columns) {
+
+        this.rows = rows;
+        this.columns = columns;
+        grid = new int[rows][columns];
+        fillGrid();
+    }
+
+    /**
+     * Llenar la matriz con 1
+     */
+    public void fillGrid() {
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < columns; j++) {
+                grid[i][j] = 1; // Fill the Grid with '1' (all white)
+            }
+        }
+    }
+
+    /**
+     * Imprimir la matriz
+     */
+    public void printGrid() {
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < columns; j++) {
+                System.out.print(grid[i][j] + "  ");
+            }
+            System.out.println();
+        }
+    }
+
+    /**
+     * Obtener la posicion en la matriz
+     */
+    public int getSquare(int x, int y) {
+        return grid[x][y];
+    }
+
+    /**
+     * Cambiar valor de la matriz
+     */
+    public void changeSquare(int x, int y) {
+        if (grid[x][y] == 1)
+            grid[x][y] = 0;
+        else
+            grid[x][y] = 1;
+    }
+
+    public int getRows() { // devuelve numero de filas
+        return grid.length;
```

Comment:
You could return `this.rows` instead. Same for `getColumns`

Reply:
The observation was taken into account.